src/main/kotlin/org/jitsi/nlj/rtp/TransportCcEngine.kt

```diff
@@ -100,37 +104,47 @@ class TransportCcEngine(
             localReferenceTime = now
         }
 
-        // We have to remember the oldest known sequence number here, as we
-        // remove from sentPacketDetails inside this loop
-        val oldestKnownSeqNum = synchronized(sentPacketsSyncRoot) { sentPacketDetails.oldestEntry() }
         for (packetReport in tccPacket) {
             val tccSeqNum = packetReport.seqNum
             val packetDetail = synchronized(sentPacketsSyncRoot) {
-                sentPacketDetails.remove(tccSeqNum)
+                sentPacketDetails.get(tccSeqNum)
             }
 
             if (packetDetail == null) {
                 if (packetReport is ReceivedPacketReport) {
+                    currArrivalTimestamp += packetReport.deltaDuration
```

Comment:
So was this a bug? (this being missing before)

Reply:
Yes, the timestamps of subsequent received packets in this report would be wrong.

Reply:
Cool, nice catch.